packages/react-instantsearch-core/src/widgets/InstantSearch.tsx

```diff
@@ -246,14 +248,15 @@ class InstantSearch extends Component<Props, State> {
     }
   }
 
-  onSearchParameters(getSearchParameters, context, props) {
+  onSearchParameters(getSearchParameters, context, props, getMetadata) {
```

Comment:
The signature becomes a bit ugly. IMO it would be simpler to introduce a new prop/function with a simpler signature. We can then adapt the current implementation on the new one without BC. Here is an example:

```js
class InstantSearch {
  onSearchParameters(widget, context, props) {
    if (this.props.onSearchParameters) {
      this.props.onSearchParameters(widget.getSearchParameters, context, props, searchState);
    }
  }

  onFindBetterNameThanThis(widget, context, props) {
    if (this.props.onFindBetterNameThanThis) {
      this.props.onFindBetterNameThanThis(widget, context, props, searchState);
    }
  }
}
```

Reply:
I didn't rename it because there's a _lot_ of people who get stuck up on forgetting to pass `onSearchParameters` down to React InstantSearch, I feel like if we change it again, it might be even worse

Reply:
cc @yannickcr @francoischalifour

Reply:
What if we warn or throw if `onSearchParameters` is missing in the context of SSR?

Reply:
these detections of server side rendering often cause accidental polyfills, not sure how to detect a prop is not passed only then

Reply:
Hmm I didn't think of accidentail polyfills. But that snippet might be okay, because it's directly coming from the internal code of React: https://github.com/facebook/react/blob/1eb2b892dfffedfdb57039ef715fc7049f4d538a/packages/shared/ExecutionEnvironment.js
but again, I'm not too sure.